File: python/bankaccount/cash_machine.py
from decimal import Decimal

from python.bankaccount.branch_finder import BranchFinder


class CashMachine:
    @property
    def branch_finder(self):
        return BranchFinder()

    def withdraw(self, town: str, customer: str, amount: Decimal):
        return self.branch_finder \
            .find_branch_for_town(town) \
            .personal_accounts() \
            .get_account_for_customer(customer) \
            .withdraw(amount)

File: go/bankaccount/cash_machine.go
package bankaccount

type CashMachine struct{}

func (c *CashMachine) BranchFinder() *BranchFinder {
	return &BranchFinder{}
}

func (c *CashMachine) Withdraw(town string, customer string, amount int) bool {
	return c.BranchFinder().
		FindBranchForTown(town).
		PersonalAccounts().
		GetAccountForCustomer(customer).
		Withdraw(amount)
}
